In [2]:
pip install memote

Note: you may need to restart the kernel to use updated packages.


In [55]:
!memote run iND750.xml

============================= test session starts =============================
platform win32 -- Python 3.11.4, pytest-7.4.0, pluggy-1.0.0 -- C:\Users\pauli\anaconda3\python.exe
cachedir: .pytest_cache
rootdir: C:\Users\pauli
plugins: anyio-3.5.0
collecting ... collected 146 items / 1 skipped

..\..\..\..\..\anaconda3\Lib\site-packages\memote\suite\tests\test_annotation.py::test_metabolite_annotation_presence FAILED [  0%]
..\..\..\..\..\anaconda3\Lib\site-packages\memote\suite\tests\test_annotation.py::test_reaction_annotation_presence FAILED [  1%]
..\..\..\..\..\anaconda3\Lib\site-packages\memote\suite\tests\test_annotation.py::test_gene_product_annotation_presence FAILED [  2%]
..\..\..\..\..\anaconda3\Lib\site-packages\memote\suite\tests\test_annotation.py::test_metabolite_annotation_overview[pubchem.compound] FAILED [  2%]
..\..\..\..\..\anaconda3\Lib\site-packages\memote\suite\tests\test_annotation.py::test_metabolite_annotation_overview[kegg.compound] FAILED [  3%]
..\..\..\..

Checking `git` installation.
The current solver interface glpk doesn't support setting the optimality tolerance.
Storing result in 'result.json.gz'.


In [139]:
!memote report diff iFF708.xml iIN800.xml iLL672.xml iMM904.xml iND750.xml yeast8.xml

Writing problem data to 'C:\Users\pauli\AppData\Local\Temp\tmpuae5w8gs.glpk'...

The current solver interface glpk doesn't support setting the optimality tolerance.
The current solver interface glpk doesn't support setting the optimality tolerance.
error: No objective coefficients in model. Unclear what should be optimized
The current solver interface glpk doesn't support setting the optimality tolerance.
The current solver interface glpk doesn't support setting the optimality tolerance.
The current solver interface glpk doesn't support setting the optimality tolerance.
The current solver interface glpk doesn't support setting the optimality tolerance.
Writing diff report to 'index.html'.



15212 lines were written
Writing problem data to 'C:\Users\pauli\AppData\Local\Temp\tmpc0xjteka.glpk'...
20822 lines were written
Writing problem data to 'C:\Users\pauli\AppData\Local\Temp\tmpvpv88f97.glpk'...
13191 lines were written
Writing problem data to 'C:\Users\pauli\AppData\Local\Temp\tmp6ikxbhwx.glpk'...
20574 lines were written
Writing problem data to 'C:\Users\pauli\AppData\Local\Temp\tmpc5sfzmn3.glpk'...
16729 lines were written
Writing problem data to 'C:\Users\pauli\AppData\Local\Temp\tmptw0m_4ud.glpk'...
49634 lines were written
============================= test session starts =============================
platform win32 -- Python 3.11.4, pytest-7.4.0, pluggy-1.0.0
rootdir: C:\Users\pauli
plugins: anyio-3.5.0
collected 146 items / 1 skipped

..\..\..\..\..\anaconda3\Lib\site-packages\memote\suite\tests\test_annotation.py F [  0%]
FFFFFFFFFF============================= test session starts =============================
platform win32 -- Python 3.11.4, pytest-7.4.0, plug

| *Saccharomyces cerevisiae* GSM model | Total score(%) | Total reactions | Total metabolites | Total Genes | Metabolic coverage | Consistency score (%) | Annotation-metabolites score  (%) | Annotation-reactions score (%) | 
|:-:|:-:|:-:|:-:|:-:|:-:|:-:|:-:|:-:| 
|iFF708| 15 | 1379 | 796 | 619 | 2.23 | 0 | 0 | 0 |
|iIN800 | 23 | 1837 | 1824 | 0 | 1.00 | 0 | 100 | 100 |
|iLL672 | 15 | 1195 | 671 | 0 | 1.00 | 0 | 0 | 0 |
|iMM904 | 31 | 1577 | 1228 | 905 | 1.74 | 100 | 0 | 0 |
|iND750 | 37 | 1266 | 1061 | 750 | 1.69 | 100 | 0 | 0 |
|yeast8| 62 | 4058 |  2742 | 1161 | 3.50 | 0 | 100 | 100 |

Number of protein genes:        6017
Number of RNA genes:             420

In [2]:
from cobra.io import read_sbml_model

In [3]:
model = read_sbml_model('yeast8.xml')

In [4]:
model

<Model yeastGEM_v8__46__7__46__1 at 0x196c9410290>

In [5]:
model.metabolites.get_by_id('s_1566')

<Metabolite s_1566 at 0x196c941c1d0>

In [6]:
model.medium

{'r_1654': 1000.0,
 'r_1714': 1.0,
 'r_1832': 1000.0,
 'r_1861': 1000.0,
 'r_1992': 1000.0,
 'r_2005': 1000.0,
 'r_2020': 1000.0,
 'r_2049': 1000.0,
 'r_2060': 1000.0,
 'r_2100': 1000.0,
 'r_4593': 1000.0,
 'r_4594': 1000.0,
 'r_4595': 1000.0,
 'r_4596': 1000.0,
 'r_4597': 1000.0,
 'r_4600': 1000.0}

In [7]:
solution = model.optimize()
solution.fluxes[solution.fluxes != 0.]

r_0005    0.061324
r_0006    0.020490
r_0007    0.006239
r_0013    0.000016
r_0014    0.000082
            ...   
r_4599    0.081928
r_4600   -0.000018
r_4702   -0.000637
r_4703   -0.000637
r_4749   -0.000016
Name: fluxes, Length: 628, dtype: float64

In [8]:
for reaction in model.metabolites.s_1566.reactions:
    print(reaction.id, reaction, reaction.name)

r_1092 r_1092: s_1198 + s_1566 --> s_0580 + s_0794 + s_1203 xylitol dehydrogenase (D-xyulose-forming)
r_2105 r_2105: s_1567 <=> s_1566 xylitol transport
r_4490 r_4490: s_0794 + s_1212 + s_4180 <=> s_1207 + s_1566 Xylitol:NADP+ 4-oxidoreductase (L-xylulose-forming)
r_1093 r_1093: s_0578 + s_0794 + s_1212 --> s_1207 + s_1566 xylose reductase


r_1093 - xylose reductase
r_1719 - D-xylose transport
r_1092 - xylitol dehydrogenase 


In [9]:
model.optimize()                               
print(model.objective.value)

0.081928158523638


In [10]:
with model:
    model.objective = model.reactions.r_1092  # Assuming XYLR is the reaction for xylitol production
    xylitol_production = model.optimize().objective_value
    print('Maximum theoretical productivity of xylitol:', xylitol_production, 'mmol/gDW*h')

In [12]:
xylose_reductase_reaction = model.reactions.get_by_id('r_1093')

# To simulate overexpression, you might increase the upper bound of the reaction.
# If you want to remove the upper limit to simulate unlimited enzyme capacity, you can set it to a very high number or to None (which COBRApy interprets as infinity).
xylose_reductase_reaction.upper_bound = 1000  # This sets the upper bound to a high number to simulate overexpression

# Optionally, if you want to ensure that the reaction only goes in the forward direction, you can set the lower bound to 0.
xylose_reductase_reaction.lower_bound = 0  # This ensures the reaction can only go in the forward direction

# After modifying the reaction bounds, you can re-optimize the model
solution = model.optimize()
print(model.objective.value)

0.081928158523638
